student.rb

```diff
@@ -1,11 +1,18 @@
 require_relative 'person'
 
 class Student < Person
+  attr_reader :classroom
+
   def initialize(classroom, age, name = 'unknown', parent_permission: true)
     super(age, name, parent_permission: parent_permission)
     @classroom = classroom
   end
 
+  def classroom=(classroom)
+    @classroom = classroom
+    owner.animals.push(self) unless owner.animals.include?(self)
```

Comment:
- [ ] As requested by the previous please use the correct classes, You're using the wrong classes provided in the examples.
  - You can consider using the code. 
  
```ruby
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end